src/encoder/varint.js

```diff
@@ -2,8 +2,22 @@
 const BN = require("bn.js")
 
 const VarInt = (signed)=> {
-  function decode () {
-    throw Error("not implemented")
+  function decode (bytes) {
```

Comment:
Pls link to the golang code where this was ported from in a comment

Reply:
https://github.com/golang/go/blob/master/src/encoding/binary/varint.go#L60